Framework/Kernel/test/VectorHelperTest.h

```diff
@@ -134,6 +134,53 @@ class VectorHelperTest : public CxxTest::TestSuite {
     TS_ASSERT_DELTA(bin_edges[2], 2.0, 1e-12);
   }
 
+  void test_flattenContainer_EmptyInputVector() {
+    const std::vector<std::vector<int>> emptyInput;
+    const auto result = VectorHelper::flattenVector<int>(emptyInput);
+    TS_ASSERT(result.empty());
+  }
+
+  void test_flattenContainer_SingleSubvectorWithMultipleValues() {
+    std::vector<std::vector<int>> input;
+    auto values = {3, 1, -1, -3, -5};
+    input.emplace_back(values);
+    const std::vector<int> expected(values);
+    const auto result = VectorHelper::flattenVector(input);
+    TS_ASSERT_EQUALS(result.size(), expected.size());
+    for (size_t i = 0; i < result.size(); ++i) {
+      TS_ASSERT_EQUALS(result[i], expected[i]);
+    }
+  }
+
+  void test_flattenContainer_MultipleSubvectorsWithSingleValues() {
+    std::vector<std::vector<int>> input;
+    input.emplace_back(std::vector<int>{3});
```

Comment:
I don't think this is doing what you expect. Using `std::vector<>::emplace_back` like this will construct a temporary, then use the copy constructor to construct a `std::vector<int>` in place. You'd be better off using the a different constructor (for example `input.emplace_back(1,3);` or using `std::vector<std::vector<int>>::push_back`.